common/unsafe/src/main/java/org/apache/spark/sql/catalyst/util/CollationFactory.java

```diff
@@ -818,7 +818,7 @@ public static UTF8String getCollationKey(UTF8String input, int collationId) {
     if (collation.supportsBinaryEquality) {
       return input;
     } else if (collation.supportsLowercaseEquality) {
-      return input.toLowerCase();
+      return CollationAwareUTF8String.toLowerCase(input);
```

Comment:
Is this needed for this PR? Also does the `toLowerCase` method produce a key where all Greek sigmas map to the same character?

Reply:
yes, we will need to fix `getCollationKey` in a separate PR (most likely this one: https://github.com/apache/spark/pull/47132), so I'll undo these changes for now